servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/TransportObserverUtils.java

```diff
@@ -27,7 +28,10 @@
 import static io.netty.util.AttributeKey.newInstance;
 import static java.util.Objects.requireNonNull;
 
-final class TransportObserverUtils {
+/**
+ * Utilities for {@link TransportObserver}.
+ */
+public final class TransportObserverUtils {
```

Comment:
I am getting increasingly concerned about the number of utilities that we are creating as part of this work. It smells like we are missing an abstraction but I can't put my finger on it yet.

Reply:
The current goal is to build the API and provide test coverage to ensure that callbacks are triggered at the correct time. Implementation intentionally uses a simplified approach to minimize internal changes until the full picture is there. The follow-up PR will reconsider the internal implementation approach.